tools/automation/src/check-plugin-updates.ts

```diff
@@ -95,18 +110,70 @@ export class Report {
             return this.handleError(entry, `Failure: there is no upstream version`);
           } else if (!entry.registryVersion) {
-            entry.needsUpdating = false;
+            entry.extensionNeedsUpdating = false;
             return this.handleError(entry, `Failure: there is no registry version`);
           } else {
-            entry.needsUpdating = semver.gt(entry.upstreamVersion, entry.registryVersion);
+            entry.extensionNeedsUpdating = semver.gt(entry.upstreamVersion, entry.registryVersion);
           }
 
           // cleanup cloned directory for this extension
           await fs.remove(entry.clonePath);
 
+          // If the extension has a sidecar image, let's check it too
+          if (extension.sidecar) {
+            const sidecarCloneName = extension.sidecar.source.repository.replace(/[^\w\s]/gi, '');
+            // Create unique clone path for the sidecar, otherwise there may be collisions
+            entry.sidecarClonePath = path.resolve(SIDECAR_ROOT_DIR, sidecarCloneName, entry.clonePath);
+
+            await fs.remove(entry.sidecarClonePath);
+            // Clone sidecar repository
+            try {
+              await git.clone(extension.sidecar.source.repository, entry.sidecarClonePath);
+            } catch (err) {
+              return this.handleError(entry, `Error cloning sidecar repository: ${err}`);
+            }
+
+            const sidecarNameSplit = extension.sidecar.image.split(':');
+            entry.sidecarLocation = `https://${sidecarNameSplit[0]}`;
+            const registryImageVersion = sidecarNameSplit[1];
+            // Filter out cases that do not adhere to the sidecar naming scheme
+            if (
+              registryImageVersion &&
+              (!registryImageVersion.includes('latest') || !registryImageVersion.includes('next'))
+            ) {
+              let sidecarSHA1;
+              entry.registrySidecarImage = registryImageVersion;
+              // Run git rev-parse --short HEAD and get hash of the upstream image
+              try {
+                const gitRevision = extension.sidecar.source.revision
+                  ? `origin/${extension.sidecar.source.revision}`
+                  : 'HEAD';
+                sidecarSHA1 = await simpleGit(entry.sidecarClonePath).revparse(['--short', `${gitRevision}`]);
+              } catch (err) {
+                return this.handleError(entry, `Error executing git rev-parse in sidecar repository: ${err}`);
+              }
+              // Compare versions
+              let upstreamSidecarImageVersion;
+              if (extension.sidecar.source.revision) {
+                upstreamSidecarImageVersion = `${extension.sidecar.source.revision}-${sidecarSHA1}`;
+                entry.sidecarNeedsUpdating = registryImageVersion !== upstreamSidecarImageVersion;
+              } else {
+                const sidecarVersion = (await fs.readFile(path.join(entry.sidecarClonePath, 'VERSION'), 'utf-8')).split(
```

Comment:
we could maybe use regexp instead of split

Reply:
Agreed, it will be a task for next week